src/fmu/dataio/providers/objectdata/_provider.py

```diff
@@ -213,7 +213,7 @@ def get_objectdata(self) -> DerivedObjectDescriptor:
             efolder="dictionaries",
             fmt=(fmt := self.dataio.dict_fformat),
             extension=self._validate_get_ext(fmt, "JSON", ValidFormats().dictionary),
-            spec=self.get_spec(),
-            bbox=self.get_bbox(),
+            spec=self.get_spec() or None,
```

Comment:
I updated value for `spec` and `bbox` to `None` for objects that doesn't have these attributes. An empty dict will trigger validation errors in pydantic since it will complain about missing fields that are not present in the dictionary.